subprojects/packagefiles/nanobind/meson.build

```diff
@@ -0,0 +1,87 @@
+project(
+    'nanobind',
+    'cpp',
+    version: '2.0.0',
+    meson_version: '>=0.53.0',
+    license: 'BSD-3-Clause',
+)
+
+py_mod = import('python')
+py = py_mod.find_installation()
+py_dep = py.dependency(embed: true)
+
+robin_map_dep = dependency('robin-map')
+
+incdir = include_directories('include')
+
+# Arguments are derived from nanobind_build_library function in CMake configuration
+dep_compile_args = []
+dep_link_args = []
+if get_option('python.allow_limited_api')
+  py_version = py.language_version()
+
+  if py_version.version_compare('>=3.12')
+    add_project_arguments('-DPy_LIMITED_API=0x030C0000', language: 'cpp')
+  endif
+endif
+
+if get_option('buildtype') == 'shared'
+  # nanobind_link_options uses linker response files for either PyPy or
+  # CPython at this step - how do we support in Meson?
+  add_project_arguments('-DNB_BUILD', language: 'cpp')
+  dep_compile_args += ['-DNB_SHARED']
+
+  if host_machine.system() == 'darwin'
+    add_project_link_arguments('-Wl,-dead_strip', '-Wl,x', '-Wl,-S', language: 'cpp')
+  elif host_machine.system() != 'windows'
+    add_project_link_arguments('-Wl,-s', language: 'cpp')
+  endif
+elif host_machine.system() != 'windows' and host_machine.system() != 'darwin'
+  dep_compile_args += ['-ffunction-sections', '-fdata-sections']
+  dep_link_args += ['-Wl,--gc-sections']
+endif
+
+# Here the CMake configuration sets PIC - does Meson handle this internally?
```

Comment:
static_library (and library, to cover static ones) has the `pic: <bool>` option. By default it is unset, which means that the value of `-Db_staticpic` is used instead, so builders can make this choice. By default, the project option is true: all static libraries will build with PIC.

You can explicitly force this with `pic: true` as a kwarg to static_library to prevent anyone from overriding it on the command line. You can also explicitly force it off with `pic: false` if you're positive you don't want to build with PIC.

In general, static libraries that may later be used by shared libraries need this, disabling PIC is a niche use case.